pick-shard.js

```diff
@@ -0,0 +1,15 @@
+// SPDX-FileCopyrightText: 2022 Mix Irving
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const bfe = require('ssb-bfe')
+const crypto = require('crypto')
+
+module.exports = function pickShard(rootFeedId, idString) {
+  const buf = Buffer.concat([bfe.encode(rootFeedId), bfe.encode(idString)])
+
+  const hash = crypto.createHash('sha256')
+  hash.update(buf)
+
+  return hash.digest('hex')[0]
```

Comment:
:fire: this always returns strings e.g. `'a'` or '`7'`
This is actually good because the `feedpurpose` for the shard MUST be a string (according to this module API anyway). Should we change the spec to clarify that it's the first hex character **as a string** that defines the shard?

Reply:
Yes, good spotting.

It could be something like

```diff
-The `content` on the root's message for the shard feed MUST have the nibble encoded as hexadecimal in the `feedpurpose` field of the `metafeed/add/derived` message.
+The `content` on the root's message for the shard feed MUST have the nibble expressed in hexadecimal and encoded as a string in the `feedpurpose` field of the `metafeed/add/derived` message.
```